src/decoder/mod.rs

```diff
@@ -1277,7 +1277,13 @@ impl<R: Read + Seek> Decoder<R> {
             return Ok(0);
         }
 
-        Ok((self.height + rows_per_strip - 1) / rows_per_strip)
+        // rows_per_strip - 1 can never fail since we know it's at least 1
+        let height = match self.height.checked_add(rows_per_strip - 1) {
+            Some(h) => h,
+            None => return Err(TiffError::IntSizeError),
```

Comment:
I'm not entirely sure if IntSizeError is the right error for here. Probably?

The places where I used LimitsExceeded is when, if the operation was on a bigint and worked, we'd try to read a huge amount of data, so I'm treating it as if the user always has a limit of `usize::MAX`.

Reply:
An `IntSizeError` seems to be appropriate.